packages/react-sdk/src/page-meta-generator.ts

```diff
@@ -0,0 +1,53 @@
+import type { Asset, DataSources, Page } from "@webstudio-is/sdk";
+
+export type PageMeta = {
+  title: string;
+  description?: string;
+  excludePageFromSearch?: boolean;
+  socialImageAssetId?: Asset["id"];
+  custom: Array<{ property: string; content: string }>;
+};
+
+export const generatePageMeta = ({
+  page,
+}: {
+  page: Page;
+  dataSources: DataSources;
+}) => {
+  const titleExpression = JSON.stringify(page.title);
+  const descriptionExpression = JSON.stringify(page.meta.description);
+  const excludePageFromSearchExpression = JSON.stringify(
+    page.meta.excludePageFromSearch
+  );
+  const socialImageAssetIdExpression = JSON.stringify(
+    page.meta.socialImageAssetId
+  );
+  let generated = "";
+  generated += `export const getPageMeta = ({}: {\n`;
```

Comment:
there really must be a better way to build that than this ... maybe fn.toString() could do this So we can write normal functions, then turn them to strings or use a parser/compiler ...

Reply:
toString would not work because code is dynamic based on data. Static functions are just imported from sdk.

Using any compiler is a hell of an overhead for node_modules. Own stuff would require maintaining.